src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,105 @@
-import React from 'react';
+import {
+  ChangeEvent,
+  Dispatch,
+  FC,
+  FormEvent,
+  SetStateAction,
+  useState,
+} from 'react';
+import cn from 'classnames';
+
+import { addComment } from '../api/comments';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
+interface Props {
+  selectedPost: Post;
+  setComments: Dispatch<SetStateAction<Comment[]>>;
+}
+
+export const NewCommentForm: FC<Props> = ({ selectedPost, setComments }) => {
+  const [authorName, setAuthorName] = useState('');
+  const [authorNameError, setAuthorNameError] = useState(false);
+  const [authorEmail, setAuthorEmail] = useState('');
+  const [authorEmailError, setAuthorEmailError] = useState(false);
+  const [comentText, setComentText] = useState('');
```

Comment:
There is a typo in the variable name `comentText`. It should be `commentText` to maintain consistency and avoid confusion.